app/bundles/PageBundle/Controller/PageController.php

```diff
@@ -880,9 +880,12 @@ public function abtestAction($objectId)
             $clone->setVariantStartDate(null);
             $clone->setIsPublished(false);
             $clone->setVariantParent($entity);
+
+            return $this->newAction($clone);
         }
 
-        return $this->newAction($clone);
+        // Given entity ID is null or invalid. Show access denied page.
+        return $this->accessDenied();
```

Comment:
Not too sure about this fix. We shouldn't get here in the first place when the given entity ID is null or invalid. Don't know which part of Mautic this function serves - prefer to have a look by an experienced Mautic dev here 🧙🏼‍♂️

Reply:
Yes, this should not happen. It's executed if an A/B test is created from a landing page. So we basically clone the original landing page and load the same values to a new form so users could make adjustments and save it.

I think that this would make more sense:
```suggestion
        return $this->notFound();
```

And ideally we should return early to make the code simpler:
```php
if (!$entity) {
    return $this->notFound();
}

// Continue with cloning the entity here.
```